src/browser/preload_scripts.ts

```diff
@@ -113,29 +113,29 @@ export function fetchAndLoadPreloadScripts(
 
 
 // resolves to type `PreloadFetched` on success
-// resolves to the input (type `PreloadInstance`) when asset fails to be cached
+// resolves to `undefined` when fetch fails to download asset to Chromium cache
 function fetch(identity: Identity, preloadScript: PreloadInstance): Promise<FetchResponse> {
     return new Promise((resolve: FetchResolver, reject: Rejector) => {
         cachedFetch(identity.uuid, preloadScript.url, (fetchError: null | Error, scriptPath: string | undefined) => {
             if (!fetchError) {
                 resolve({identity, preloadScript, scriptPath});
             } else {
                 updatePreloadState(identity, preloadScript, 'load-failed');
-                reject();
+                resolve();
```

Comment:
well, no, not originally, because originally we were using a different paradigm.

### Originally we had this:
1. Fetch (to Chromium cache) & load (from Chromium cache) all preload scripts via resource fetcher _for each and every page load_
2. Wait for all to complete loading
3. Proceed to `loadURL` _if and only if_ all "required" (not optional) scripts loaded successfully
4. Eval loop in api_decorator executes all available scripts
5. Scripts are released from memory after eval

So `reject` was being called on `if (!optional)` when a required script failed to load, and this call was an overlooked remnant from that code.

### We then changed to this:
1. Fetch & load all of them via resource fetcher _if and only if_ not previously fetched, caching scripts in memory (as a side-effect) as they are successfully loaded
3. Wait for any new scripts to complete loading
4. Proceed to `loadURL` regardless of load success
5. Eval loop in api_decorator executes all successfully loaded scripts _if and only if_ that set includes all required scripts
6. Scripts remain in memory for subsequent page loads (any app, any window)

#### Advantages of new paradigm:
1. Our resource fetcher currently not smart enough to skip re-fetching external files over and over again, so maintaining an in-memory cache gives us simple control of that
2. Skips reloading locally from disc cache on every page (minor impact though that is)

#### Disadvantage of new paradigm:
1. Uses more memory (for scripts), judged to be minor impact
2. Edited scripts are not refetched until runtime restarts

We made one final change to the above when Omer pointed out that local files (`file://`) are never actually fetched, but load directly from their native disc locations. #147 changes the logic to always reload local files so app developers can freely edit them and expect to see different results without restarting runtime (however see #147 for limitations on this).